Demo/Sources/Showcase/SourceCodeViewable.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  Copyright (c) SRG SSR. All rights reserved.
+//
+//  License information is available from the LICENSE file.
+//
+
+import SwiftUI
+
+private struct SourceCodeButton: View {
+    let action: () -> Void
+
+    var body: some View {
+        Button(action: action) {
+            HStack {
+                Text("GitHub")
+            }
+        }
+    }
+}
+
+protocol SourceCodeViewable {
+    static var filePath: String { get }
+}
+
+extension View {
+    @ViewBuilder
+    func sourceCode<T>(of objectType: T.Type?) -> some View where T: SourceCodeViewable {
```

Comment:
Why is an optional introduced here? This fact is not used anywhere in the code and I would always expect to have a type supplied here.